plenum/server/domain_req_handler.py

```diff
@@ -163,32 +164,49 @@ def getNymDetails(state, nym, isCommitted: bool=True):
     def nym_to_state_key(nym: str) -> bytes:
         return sha256(nym.encode()).digest()
 
-    def make_proof(self, path, head_hash=None):
+    def make_proof(self, path, head_hash=None, get_value=False):
```

Comment:
I don't like that we return value in a method called `make_proof`. I think we need more re-factorings to support this change properly. 
A better way could be to return the proof during lookup (when calling get).

Reply:
> A better way could be to return the proof during lookup (when calling get).

This is exactly what is mentioned in the PR comment. But we should not be living with this to wait for such refactoring.

Reply:
I don't think that renaming a method is a very big change.

Reply:
I was not talking about renaming `make_proof`, the trie has a `get` method, that should be augmented to have a `with_proof` boolean parameter (`False` by default). Then this get method is used throughout the code and we get rid of `produce_spv_proof`.

Reply:
It may be OK to not touch it for a while in a tree (this was originally not our code, so maybe this is OK).
But I don't like that we doing the same confusing things in our API (`make_proof`, `req_handlers`'s `lookup` methods)

Reply:
I think we can move creation of proof to the place where we get the actual result (lookup for example), and pass `proof` to `make_proof` (renaming `make_proof` to `make_proof_result`) as an argument, so that `make_proof` is responsible for just creation of a result data structure containing proof, multi_sig, etc.